src/libflake/flake/flake.cc

```diff
@@ -99,7 +99,7 @@ static std::map<FlakeId, FlakeInput> parseFlakeInputs(
     const std::optional<Path> & baseDir, InputPath lockRootPath);
 
 static FlakeInput parseFlakeInput(EvalState & state,
-    std::string_view inputName, Value * value, const PosIdx pos,
+    std::optional<std::string_view> inputName, Value * value, const PosIdx pos,
```

Comment:
Could  `inputName` be removed? An input shouldn't have to know its name.

I think this might be an issue with the data model, but it seems that we have low hanging fruit, in that we could turn the call *with* `inputName` into a separate overload that calls `parseFlakeInput(state,value,pos)` and postprocesses the return value.